components/camel-jta/src/main/java/org/apache/camel/jta/JtaTransactionErrorHandler.java

```diff
@@ -64,4 +69,16 @@ public ErrorHandler clone(Processor output) {
         return answer;
     }
 
+    @Override
+    public boolean process(final Exchange exchange, final AsyncCallback callback) {
+        if (!exchange.isTransacted()) {
+            LOG.debug("mark " + exchange + " transacted");
+            exchange.getUnitOfWork().beginTransactedBy("jta");
+            super.process(exchange, callback);
+            exchange.getUnitOfWork().endTransactedBy("jta");
```

Comment:
I think you need to have the endTransacted in a try .. finally block to ensure it will be ended